File: android/src/main/kotlin/com/gdelataillade/alarm/models/TimePressureSettings.kt
package com.gdelataillade.alarm.models

import com.gdelataillade.alarm.generated.TimePressureSettingsWire
import kotlinx.serialization.Serializable

@Serializable
data class TimePressureSettings(
    val enable: Boolean,
    val volume: Double,
    val speechRate: Double,
    val pitch: Double,
    val loop: Boolean,
    val loopInterval: Long,
) {
    companion object {
        fun fromWire(e: TimePressureSettingsWire): TimePressureSettings {
            return TimePressureSettings(
                e.enable,
                e.volume,
                e.speechRate,
                e.pitch,
                e.loop,
                e.loopInterval,
            )
        }
    }
}

File: ios/Classes/models/TimePressureSettings.swift
import Foundation

struct TimePressureSettings: Codable {
    let enable: Bool
    let volume: Double
    let speechRate: Double
    let pitch: Double
    let loop: Bool
    let loopInterval: Int64
    let languageTag: String?

    static func from(wire: TimePressureSettingsWire) -> TimePressureSettings {
        return TimePressureSettings(
            enable: wire.enable,
            volume: wire.volume,
            speechRate: wire.speechRate,
            pitch: wire.pitch,
            loop: wire.loop,
            loopInterval: wire.loopInterval,
            languageTag: wire.languageTag
        )
    }
}
